h2o-core/src/main/java/water/ClientDisconnectCheckThread.java

```diff
@@ -0,0 +1,43 @@
+package water;
```

Comment:
Is this another listener like `ListenerService`?

Reply:
I wouldn't say this is a listener. This is a basically a simple thread checking whether the client has been disconnected or not.

But we might actually store the history of joined/disconnected clients using the listenerservice to keep track of it.